apps/web/src/pages/Applications/ApplicationProfilePage/ApplicationProfilePage.tsx

```diff
@@ -75,16 +100,170 @@ export const getPageInfo: GetApplicationPageInfo = ({
+
+  const handleNavigation = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault(); // We don't want to navigate until we mark the step as complete
+
+    if (!pageInfo.hasError) {
+      executeSubmitMutation({
+        id: application.id,
+        application: {
+          insertSubmittedStep: ApplicationStep.Welcome,
+        },
+      })
+        .then((res) => {
+          if (res.data) {
+            navigate(nextStepPath);
+          }
+        })
+        .catch(() => {
+          toast.error(
+            intl.formatMessage(errorMessages.unknownErrorRequestErrorTitle),
+          );
+        });
+    } else {
+      toast.error(
+        intl.formatMessage({
+          defaultMessage:
+            "Please complete all required fields before continuing.",
+          id: "G1jegJ",
+          description:
+            "Error message displayed when user attempts to submit incomplete profile",
+        }),
+      );
+    }
+
+    return false;
+  };
 
-  return <Heading data-h2-margin-top="base(0)">{pageInfo.title}</Heading>;
+  const handleUpdate: SectionProps["onUpdate"] = (userId, userData) => {
+    return executeUpdateMutation({
+      id: userId,
+      user: userData,
+    }).then((res) => res.data?.updateUserAsUser);
+  };
+
+  const sectionProps = {
+    user,
+    isUpdating,
+    onUpdate: handleUpdate,
+  };
+
+  return (
+    <>
+      <Heading data-h2-margin-top="base(0)">{pageInfo.title}</Heading>
+      <p>
+        {intl.formatMessage({
+          defaultMessage:
+            "This step includes all of your profile information required for this application. Each section has a series of required fields that must be completed to move on to the next step. Completing this profile information will only happen once, and the information you provide here will auto-populate on applications you submit moving forward.",
+          id: "D9Elm9",
+          description: "Application step to complete your profile, description",
+        })}
+      </p>
+      <div
+        data-h2-margin="base(x2, 0)"
+        data-h2-display="base(flex)"
+        data-h2-flex-direction="base(column)"
+        data-h2-gap="base(x1, 0)"
+      >
+        <PersonalInformation {...sectionProps} />
+        <WorkPreferences {...sectionProps} />
+        <DiversityEquityInclusion {...sectionProps} />
+        <GovernmentInformation {...sectionProps} />
+        <LanguageProfile {...sectionProps} />
+      </div>
+      <Separator
+        orientation="horizontal"
+        data-h2-background-color="base(gray.lighter)"
+        data-h2-margin="base(x2, 0)"
+        decorative
+      />
+      <div
+        data-h2-display="base(flex)"
+        data-h2-gap="base(x.25, x.5)"
+        data-h2-flex-wrap="base(wrap)"
+        data-h2-flex-direction="base(column) l-tablet(row)"
+        data-h2-align-items="base(flex-start) l-tablet(center)"
+      >
+        <form onSubmit={handleNavigation}>
```

Comment:
can't move on to the next step and not sure why or what is going wrong

![gif](http://g.recordit.co/Vcilyy4WIy.gif)

Reply:
Okay, I think this was because language is a little more complex than just filling out required info (need to match language requirements).

Fixed in [fix profile step validation for language](https://github.com/GCTC-NTGC/gc-digital-talent/pull/6296/commits/0088bb6ff4c973e2480c1f5de64a24477dab751a)

Reply:
We only care about bilingual reqiuirements (not entirely sure why) but `FRENCH` and `ENGLISH` will never cause an error despite what the user has filled out. 🤷‍♀️ 

REF: https://github.com/GCTC-NTGC/gc-digital-talent/pull/5557#issuecomment-1413814469

Reply:
well if that is how it is supposed to work then resolved I guess 🤷